packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -505,29 +501,52 @@ export const fetchSearchIndexes = (): IndexesThunkAction<
       return;
     }
 
-    if (!dataService || !dataService.isConnected()) {
-      debug('warning: trying to load indexes but dataService is disconnected');
+    const notFetchableStatuses: SearchIndexesStatus[] = [
```

Comment:
Probably this could be extracted out, into a global constant, so it's not recreated on every call. Not too expensive and the runtime could optimise it, but the scoping seems more clear.

Reply:
added in [13d92db](https://github.com/mongodb-js/compass/pull/4883/commits/13d92db4ddd4b56db150eea6594cb8febdc1182c)